buildtools/publish.js

```diff
@@ -75,3 +107,23 @@ function replaceUrl(replacementUrl, configUrl) {
     replacementUrl
   );
 }
+
+
+function listSubmodules(repoRoot) {
+  const gitmodulesPath = path.join(repoRoot, '.gitmodules');
+  if (!fs.existsSync(gitmodulesPath)) {
+    return []; // No submodules
+  }
+
+  const content = fs.readFileSync(gitmodulesPath, 'utf-8');
+  const submodulePaths = [];
+
+  // Parse lines like: path = some/path
+  const pathRegex = /^\s*path\s*=\s*(.+)$/gm;
+  let match;
+  while ((match = pathRegex.exec(content)) !== null) {
+    submodulePaths.push(match[1]);
+  }
+
+  return submodulePaths;
+}
```

Comment:
Can we shell out to git here? We already shell out elsewhere. This gives me more confidence than a regexp that might not cover some unusual git syntax.

```suggestion
function listSubmodules(repoRoot) {
  const gitmodulesPath = path.join(repoRoot, '.gitmodules');
  if (!fs.existsSync(gitmodulesPath)) {
    return []; // No submodules
  }

  const stdout = execSync(
    'git config --file .gitmodules --get-regexp path',
    { cwd: repoRoot, encoding: 'utf-8' }
  );
  return stdout.split('\n').map(line => line.split(' ')[1]);
}
```